openedx/core/djangoapps/course_groups/models.py

```diff
@@ -185,14 +185,35 @@ class CourseCohortsSettings(models.Model):
     @property
     def cohorted_discussions(self):
         """Jsonify the cohorted_discussions"""
-        return json.loads(self._cohorted_discussions)
+
+        segmented_discussions = self.segmented_discussions.all()
```

Comment:
Actually, I'm changing my mind about this comment-- I think that is actually MY story. You just need to create the new model and patch it in, which is what you have done.